### ISSUE TITLE
The implementation of Eq.(5)

### ISSUE THREAD

**Issue by LiRunyi2001** [yunqing-me/WatermarkDM#11]
Hi there! I am very interested in this brilliant work, and I am curious about the implementation of this equation in watermarking stable diffusion:
![image](https://github.com/yunqing-me/WatermarkDM/assets/54633216/f4706efa-70e0-4840-8dda-b65ba5ab5a1b)

I would be appreciate it if there could be any explanation or guidance. Thanks in advance!

**Comment by yunqing-me** [comment]
Hi, thanks for your interest in our work!

- For the first term, it is a normal denoising loss in Diffusion Model. The goal of this term is to enable the Stable Diffusion model learn to generate the target image ($\widetilde{x}$) given the text condition ($\widetilde{c}$). Note that the text condition $\widetilde{c}$ is thus the "trigger prompt" mentioned in our paper. Through this term, we can "plant" the trigger prompt and the watermark image into the Stable diffusion
- For the second term, since we aim to preserve the original pretrained performance of Stable Diffusion, we add this regularization term to penalize the weights change during fine-tuning (using the first term in Eqn.5).

In short, the overall principle is to "plant the watermark into Stable Diffusion, while still preserving the pretrained performance".

Hope this helps you.
